packages/jupyter-widgets/__tests__/manager/manager.spec.ts

```diff
@@ -1,25 +1,90 @@
 import { IntSliderView } from "@jupyter-widgets/controls";
 import { Map } from "immutable";
 
+import { ManagerActions } from "../../src/manager/index";
 import { WidgetManager } from "../../src/manager/widget-manager";
+import * as customWidgetLoader from "../../src/manager/widgetLoader";
+
+// A mock valid module representing a custom widget
+const mockFooModule = {
+  "foo" : "bar"
+};
+// Mock implementation of the core require API
+const mockRequireJS = jest.fn((modules, ready, errCB) => ready(mockFooModule));
+(window as any).requirejs = mockRequireJS;
+(window as any).requirejs.config  = jest.fn();
+
+// Manager actions passed as the third arg when instantiating the WidgetManager class
+const mockManagerActions: ManagerActions["actions"] = {
+  appendOutput: jest.fn(),
+  clearOutput: jest.fn(),
+  updateCellStatus: jest.fn(),
+  promptInputRequest: jest.fn()
+};
+
 
 describe("WidgetManager", () => {
   describe("loadClass", () => {
+    beforeAll(() => {
```

Comment:
is this `jest.clearAllMocks()` needed?

Reply:
Yes, I think we need it. Since we have strict assertions on the number of times a mock function is called, we clear all mocks which essentially clears the transient state of a mock function like its call args and other meta data so that we start clean after every test.